### PR TITLE
[Feature]Support mask cloud credential in show create catalog and show create file table

### LINKED ISSUE #19125: Support mask aws access_key and secret_key when show create external catalog and show create file table
## Feature request
Support mask aws access_key and secret_key when show create external catalog and show create file table

### PR DESCRIPTION
**Pull request by leoyy0316**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19125

## Problem Summary(Required) ：

Support mask aws access_key and secret_key when show create external catalog and show create file table

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by leoyy0316** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java [hunk 1]
```diff
@@ -235,4 +236,16 @@ public static int calBucketNumAccordingToBackends() {
         }
         return bucketNum;
     }
+
+    public static void maskProperties(Map<String, String> properties) {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java:240]: maskCloudCredentialProperties

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java [hunk 2]
```diff
@@ -235,4 +236,16 @@ public static int calBucketNumAccordingToBackends() {
...
+        // aws.s3.access_key -> ******
+        properties.computeIfPresent(CloudConfigurationConstants.AWS_S3_ACCESS_KEY, (k, v) -> "******");
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java:242]: Extract a common function to mask, like `maskCloudCredential()` , we will need to mask like azure, gcp in the future.

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java [hunk 3]
```diff
@@ -235,4 +236,16 @@ public static int calBucketNumAccordingToBackends() {
...
+        // aws.s3.secret_key -> ******
+        properties.computeIfPresent(CloudConfigurationConstants.AWS_S3_SECRET_KEY, (k, v) -> "******");
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java:244]: I recommend masking part of characters, it's better for a user to check credentials.
For example, original `ak=efi923#31` => `ak=ef**31`.

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java (hunk 2 above)
Comment (leoyy0316) [reply fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java]: i think maskCloudCredential can use to mask all ,just to add:
properties.computeIfPresent(CloudConfigurationConstants.AZURE_ACCESS_KEY, (k, v) -> "******");

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java (hunk 3 above)
Comment (leoyy0316) [reply fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java]: > I recommend masking part of characters, it's better for a user to check credentials. For example, original `ak=efi923#31` => `ak=ef**31`.

done

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java [hunk 4]
```diff
@@ -235,4 +236,23 @@ public static int calBucketNumAccordingToBackends() {
...
+        // aws.glue.access_key -> ******
+        properties.computeIfPresent(CloudConfigurationConstants.AWS_GLUE_ACCESS_KEY,
+                (k, v) -> replaceWithIndex(2, v.length() - 2, v, "******"));
+        // aws.glue.secret_key -> ******
+        properties.computeIfPresent(CloudConfigurationConstants.AWS_GLUE_SECRET_KEY,
+                (k, v) -> replaceWithIndex(2, v.length() - 2, v, "******"));
+    }
+
+    public static String replaceWithIndex(int start, int end, String oldChar, String replaceChar) {
+        return String.valueOf(new StringBuilder(oldChar).replace(start, end, replaceChar));
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java:256]: If oldChar's length is less than 2, what will happen?

**Review by Smith-Cruise — COMMENTED** [review]
Remains LGTM, it's better if you can add a UT.

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Review by Smith-Cruise — DISMISSED** [review]
Plz, change PR's title to `[Feature] support to mask cloud credential in show create catalog`.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 16 (62.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|[2584, 2585, 2590]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|9|12|75.00%|[251, 254, 259]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java [hunk 5]
```diff
@@ -235,4 +236,26 @@ public static int calBucketNumAccordingToBackends() {
         }
         return bucketNum;
     }
+
+    public static void maskCloudCredential(Map<String, String> properties) {
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java:240]: You can create a new file into credential package, maybe named `CloudCredentialUtil.java`, then put this static method into here.😃

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 17 (58.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|[2584, 2585, 2590]|
|:large_blue_circle:|com/starrocks/credential/CloudCredentialUtil.java|9|13|69.23%|[19, 29, 32, 37]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|1|100.00%|[]|

**Review by Smith-Cruise — DISMISSED** [review]
Thks!

File: fe/fe-core/src/main/java/com/starrocks/credential/CloudCredentialUtil.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        properties.computeIfPresent(CloudConfigurationConstants.AWS_GLUE_ACCESS_KEY,
+                (k, v) -> replaceWithIndex(2, v.length() - 2, v, "******"));
+        // aws.glue.secret_key -> ******
+        properties.computeIfPresent(CloudConfigurationConstants.AWS_GLUE_SECRET_KEY,
+                (k, v) -> replaceWithIndex(2, v.length() - 2, v, "******"));
+    }
+
+    public static String replaceWithIndex(int start, int end, String oldChar, String replaceChar) {
+        if (start > end) {
+            return "******";
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/credential/CloudCredentialUtil.java:37]: make it static const

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Review by Smith-Cruise — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 17 (58.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|3|00.00%|[2584, 2585, 2590]|
|:large_blue_circle:|com/starrocks/credential/CloudCredentialUtil.java|9|13|69.23%|[19, 31, 34, 39]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|1|100.00%|[]|

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Youngwb — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.0

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-2.5

**Comment by Smith-Cruise** [comment]
Backport to branch-2.5 failed, plz cherry-pick manually.

**Comment by leoyy0316** [comment]
> Backport to branch-2.5 failed, plz cherry-pick manually.

@Smith-Cruise  ok, will do it.

**Comment by Smith-Cruise** [comment]
> > Backport to branch-2.5 failed, plz cherry-pick manually.
> 
> @Smith-Cruise ok, will do it.

Notice, reference to this pr, so we can track it easily.

**Pull request merged**